src/form-engine.component.tsx

```diff
@@ -176,7 +176,11 @@ const FormEngine = ({
                       }}>
                       {mode === 'view' ? t('close', 'Close') : t('cancel', 'Cancel')}
                     </Button>
-                    <Button type="submit" disabled={isLoadingDependencies || mode === 'view'}>
+                    <Button
+                      className={styles.saveButton}
+                      disabled={isLoadingDependencies || !isSubmitting || mode === 'view'}
```

Comment:
Fix the failing tests by:

```suggestion
                      disabled={isLoadingDependencies || isSubmitting || mode === 'view'}
```